src/main/java/com/conveyal/gtfs/model/Stop.java

```diff
@@ -125,4 +202,227 @@ public Iterator<Stop> iterator() {
             return feed.stops.values().iterator();
         }   	
     }
+
+    /**
+     * Merge stop areas into stops when loading from file.
+     */
+    public static void getCsvReaderForStopsWithStopAreas(Map<String, Stop> stops, Map<String, StopArea> stopAreas) {
+        Map<String, Set<String>> stopAreasByStopId = new HashMap<>();
+
+        stopAreas.values().forEach(stopArea ->
+            stopAreasByStopId
+                .computeIfAbsent(stopArea.stop_id, id -> new HashSet<>())
+                .add(stopArea.area_id)
+        );
+
+        stops.values().forEach(stop -> stop.stop_area_ids = getStopAreaIds(stopAreasByStopId, stop.stop_id));
+    }
+
+    /**
+     * Merge stop areas into stops when loading into DB.
+     */
+    public static CsvReader getCsvReaderForStopsWithStopAreas(
+        CsvReader stopsReader,
+        Map<String, Set<String>> stopAreasByStopId
+    ) {
+        List<String> rows = new ArrayList<>();
+        try {
+            while (stopsReader.readRecord()) {
+                String stopId = stopsReader.get(STOP_ID_FIELD);
+                rows.add(createRow(stopsReader, getStopAreaIds(stopAreasByStopId, stopId)));
+            }
+            return (rows.isEmpty())
+                ? stopsReader
+                : produceCsvPayload(rows, CSV_HEADER_FOR_MERGE);
+        } catch (Exception e) {
+            LOG.error("Error while merging stops", e);
+            // Any issues, return the original stops reader (minus stop areas).
+            return stopsReader;
+        }
+    }
+
+    /**
+     * Get all stop areas matching provided stop id.
+     */
+    private static String getStopAreaIds(Map<String, Set<String>> stopAreasByStopId, String stopId) {
+        return Optional.ofNullable(stopAreasByStopId.get(stopId))
+            .map(areas -> String.join(STOP_AREAS_SEPARATOR, areas))
+            .orElse("");
+    }
+
+    /**
+     * Create a CSV row of original stop fields plus the stop area ids.
+     */
+    private static String createRow(CsvReader stopsReader, String stopAreaIds) throws IOException {
+        String[] fields = new String[CSV_FIELDS.length];
+        for (int i = 0; i < CSV_FIELDS.length; i++) {
+            fields[i] = stopsReader.get(CSV_FIELDS[i]);
+        }
+        return String.join(",", fields) + "," + stopAreaIds;
```

Comment:
Nit: I got a little confused with this version of `createRow` that does not include a new line, whereas the other one includes a new line.